### ISSUE TITLE
Disable Mac help topics in Help menu

### ISSUE THREAD

**Issue by roblourens** [Microsoft/vscode#18697]
If you search in the Help menu, it will show you general Mac help topics:

![image](https://cloud.githubusercontent.com/assets/323878/22047411/c415c0d8-dcdb-11e6-93de-17f3f911d098.png)

None of these are from Code, but could be confused with Code-specific help, especially for certain search terms. Is it possible to disable this feature?

![image](https://cloud.githubusercontent.com/assets/323878/22047435/ee36f896-dcdb-11e6-9dee-c6a704e21ddf.png)

**Comment by bpasero** [comment]
@roblourens please report against electron.

**Comment by roblourens** [comment]
https://github.com/electron/electron/issues/8431

**Comment by roblourens** [comment]
Response:

> Looks like this can be configured in the .plist file, from my testing if you add an any value then the default help topics aren't displayed.
>
> Example taken from https://developer.apple.com/library/content/documentation/Carbon/Conceptual/ProvidingUserAssitAppleHelp/registering_help/registering_help.html#//apple_ref/doc/uid/TP30000903-CH207-CHDGHHDF
> ```
> <key>CFBundleHelpBookFolder</key>
> <string>SurfWriter.Help</string>
> <key>CFBundleHelpBookName</key>
> <string>com.mycompany.surfwriter.help</string>
> ```

It looks like this can be configured in the info.plist file in an app's package. Not sure where that happens in our code.

**Comment by bpasero** [comment]
@joaomoreno might know if we can add this to our plist

**Comment by shnaramn** [comment]
I would like to contribute. But I'm relatively new to OSS/git/VSCode. I might take some time/need help, but can I try fixing this?

**Comment by joaomoreno** [comment]
Sure.

**Comment by JeremyLoy** [comment]
Found the root cause of the issue.

TL;DR: This is blocked and needs to be flagged 'upstream' 

The build process for VS Code uses the plugin ['gulp-atom-electron'](https://github.com/joaomoreno/gulp-atom-electron), which unfortunately doesn't expose the ability to alter the Info.plist. See [here](https://github.com/joaomoreno/gulp-atom-electron/blob/master/src/darwin.js#L61)

Once the upstream plugin has been patched, it will be trivial to fix this issue.
@roblourens already found the answer; simply adding the keys `<key>CFBundleHelpBookFolder</key>
<string>SurfWriter.Help</string>
<key>CFBundleHelpBookName</key>
<string>com.mycompany.surfwriter.help</string>` to the info.plist removes the macOS help suggestions.

The author of 'gulp-atom-electron', @joaomoreno is in this thread, so if I am wrong, please correct me!

Until then though, I move to mark this as blocked/upstream and I will move on to something else.  Boy was it fun tracking this down though!

**Comment by joaomoreno** [comment]
Not so sure about that fix... Those keys seem too specific to the SurfWriter application.

**Comment by joaomoreno** [comment]
@JeremyLoy What values should we put to disable the menu?
